rust/connlib/libs/common/src/messages.rs

```diff
@@ -89,8 +98,11 @@ pub struct ReuseConnection {
 // Custom implementation of partial eq to ignore client_rtc_sdp
 impl PartialEq for RequestConnection {
     fn eq(&self, other: &Self) -> bool {
+        use secrecy::ExposeSecret;
+
         self.resource_id == other.resource_id
-            && self.client_preshared_key == other.client_preshared_key
+            && self.client_preshared_key.expose_secret()
+                == other.client_preshared_key.expose_secret()
```

Comment:
We should remove this here too.

Reply:
👍

Reply:
Removed. Comparing ResourceIds (which themselves are UUIDv4s) should be enough, especially since this should only contain clients/resources in a single Firezone Account.